File: TrackMyAct.Server/Models/Entities/OrganizerEntity.cs
namespace TrackMyAct.Server.Models.Entities;

public class OrganizerEntity : UserEntity
{
    public string Organization { get; set; }
    public string Key { get; set; }
    public ICollection<EventEntity> OrganizedEvents { get; set; }
    public ICollection<EventEntity> ArchiveEvents { get; set; }
    public ICollection<ParticipantEntity> Subscribers { get; set; }
    public ICollection<CommentEntity> Comments { get; set; }
}

File: TrackMyAct.Mobile/composeApp/src/commonMain/kotlin/org/js/tma/dto/OrganizerEntity.kt
package org.js.tma.dto

import kotlinx.serialization.Serializable

@Serializable
data class OrganizerEntity(
    override var id: Int,
    override var username: String,
    override var email: String,
    override var passwordHash: String,
    override var name: String,
    override var roleType: String,

    var organization: String,
    var key: String,
    var organizedEvents: List<EventEntity> = emptyList(),
    var archiveEvents: List<EventEntity> = emptyList(),
    var subscribers: List<ParticipantEntity> = emptyList(),
    var comments: List<CommentEntity> = emptyList()
) : UserEntity()